source/set-optional.d.ts

```diff
@@ -32,6 +34,6 @@ export type SetOptional<BaseType, Keys extends keyof BaseType> =
 		// Pick just the keys that are readonly from the base type.
 		Except<BaseType, Keys> &
 		// Pick the keys that should be mutable from the base type and make them mutable.
-		Partial<Except<BaseType, Exclude<keyof BaseType, Keys>>>
+		Partial<DistributedPick<BaseType, Keys & KeysOfUnion<BaseType>>>
```

Comment:
Would not require intersection with `KeysOfUnion<BaseType>` once #1009 is finalised.

cc: @Emiyaaaaa